[Durable Objects] Document alarm feature (#9165)
cloudflare/cloudflare-docs: content/workers/runtime-apis/durable-objects.md

The method takes no parameters, does not return a result, and can be `async`.

> #### Use `alarm()` handler method

Reviewer: Maybe just `#### Example` or `#### How to use` here?

In your Durable Object, the `alarm()` handler will be called when the alarm executes. Call `state.storage.setAlarm()` from anywhere in your Durable Object, and pass in a time for the alarm to run at. Use `state.storage.getAlarm()` to retrieve the currently set alarm time.

> The example below will show you how to implement an `alarm()` handler that wakes the Durable Object up once every 10 seconds to batch requests to a single Durable Object. The `alarm()` handler will delay processing until there is enough work in the queue for it to be worth processing them.

Reviewer: ```suggestion
The example below implements an `alarm()` handler that wakes the Durable Object up once every 10 seconds to batch requests to a single Durable Object. The `alarm()` handler will delay processing until there is enough work in the queue.
```

The example below will show you how to implement an `alarm()` handler that wakes the Durable Object up once every 10 seconds to batch requests to a single Durable Object. The `alarm()` handler will delay processing until there is enough work in the queue for it to be worth processing them.

```js
export default {
  async fetch(request, env) {
    let id = env.BATCHER.idFromName("foo");
    return await env.BATCHER.get(id).fetch(request);
  },
};

const SECONDS = 1000;

export class Batcher {
  constructor(state, env) {
    this.state = state;
    this.storage = state.storage;
    this.state.blockConcurrencyWhile(async () => {
      let vals = await this.storage.list({ reverse: true, limit: 1 });
      this.count = vals.size == 0 ? 0 : parseInt(vals.keys().next().value);
    });
  }
  async fetch(request) {
    this.count++;

    // If there is no alarm currently set, set one for 10 seconds from now

>     // Any further POSTs in the next 10 seconds will be part of this kh.

Reviewer 2: "Any further POSTs in the next 10 seconds will be part of this **batch**."

}
```

> The `alarm()` handler will be called once every 10 seconds. If an unexpected error terminates the Durable Object, the `alarm()` handler will be re-instantiated on another machine. Following a short delay, the `alarm()` handler can continue processing on the other machine.

Reviewer: ```suggestion
The `alarm()` handler will be called once every 10 seconds. If an unexpected error terminates the Durable Object, the `alarm()` handler will be re-instantiated on another machine. Following a short delay, the `alarm()` handler will run from the beginning on the other machine.
```

> The `alarm()` handler will be called once every 10 seconds. If an unexpected error terminates the Durable Object, the `alarm()` handler will be re-instantiated on another machine. Following a short delay, the `alarm()` handler can continue processing on the other machine.

Author: If an unexpected error terminates the Durable Object, `alarm()` will start from the beginning somewhere else. 

So, it sounds like there's actually **_nothing_** that continues processing here. If there's an error, the alarm will start from the beginning, but somewhere else (where exactly, though? Is it just another machine?)
